lib/plug/ip_whitelist/environment_variable_parser.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Plug.IpWhitelist.EnvironmentVariableParser do
+  @moduledoc """
+    IP Whitelists often need to be stored as an environment variable.
+    This module provides functionality for parsing a list of IP ranges out of a
+    string which could be stored in your application's environment variables.
+    Example:
+      "1.1.1.1-1.1.1.1 2.3.4.5-6.7.8.9"
+  """
+
+  @doc """
+    Given a string of ip ranges, convert to a list of ip ranges
+    Example input:
+      "1.1.1.1-1.1.1.1 2.3.4.5-6.7.8.9"
+    Example output:
+      [
+        {{1, 1, 1, 1}, {1, 1, 1, 1}},
+        {{2, 3, 4, 5}, {6, 7, 8, 9}}
+      ]
+  """
+  @spec parse(String.t()) :: [IpWhitelist.ip_range(), ...]
+  def parse(string_ip_whitelist) do
+    string_ip_whitelist
+    |> String.split(" ")
+    |> Enum.map(&parse_ip_range/1)
+  end
+
+  @doc """
+    Given a string which contains an ip range, convert to an ip_range
+    Example input:
+      1.1.1.1-1.1.1.1
+    Example output:
+      {{1, 1, 1, 1}, {1, 1, 1, 1}}
+  """
+  @spec parse_ip_range(String.t()) :: IpWhitelist.ip_range()
+  def parse_ip_range(string_ip_range) do
+    string_ip_range
+    |> String.split("-")
+    |> Enum.map(&parse_ip_address/1)
+    |> List.to_tuple()
+  end
+
+  @doc """
+    Given a string which contains an ip address, convert to an ip_address
+    Example input:
+      "1.1.1.1"
+    Example output:
+      {1, 1, 1, 1}
+  """
+  @spec parse_ip_address(String.t()) :: IpWhitelist.ip_address()
+  def parse_ip_address(string_ip_address) do
```

Comment:
What (if any) considerations are needed for IPv6? I doubt we'll need to think about that for a while at least. Right?

Reply:
Correct -  not even sure that one could have a static IPv6 address yet, but I don't know a lot about that. All of the ip addresses we've had to whitelist so far are IPv4, but I think that would be a good feature to add once it is needed

Reply:
Ok cool, figures. I too haven't seen a public IPv6 (ever?).